packages/server/src/auth.ts

```diff
@@ -67,4 +64,21 @@ export class ServerAuth {
   }
 }
 
+export const parseBasicAuth = (
+  token: string,
+): { username: string; password: string } | null => {
+  if (!token.startsWith(BASIC)) return null
+  const b64 = token.slice(BASIC.length)
+  let parsed: string[]
+  try {
+    parsed = uint8arrays
+      .toString(uint8arrays.fromString(b64, 'base64pad'), 'utf8')
+      .split(':')
+  } catch (err) {
+    return null
+  }
```

Comment:
What's the error that could happen here?

Also, a small/unrelated thing, but I wonder if it may be interesting to consider incorporating something like [bounce](https://hapi.dev/module/bounce/) to distinguish unrecoverable node/"system" errors (e.g. TypeError, SyntaxError) from operational ones.

Reply:
Invalid base64 throws here. `abc%%|%|%|%|%` would throw for instance. and since i'm using in the log serializer, it's probably best to catch it & not let it bubble up

Reply:
ah nice I like the look of something like `bounce` for sure